static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/PostTransformer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+import java.lang.instrument.ClassFileTransformer;
+import java.security.ProtectionDomain;
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+public class PostTransformer implements ClassFileTransformer {
+  @Override
+  @Nullable
+  public byte[] transform(
+      ClassLoader loader,
+      String className,
+      Class<?> classBeingRedefined,
+      ProtectionDomain protectionDomain,
+      byte[] classfileBuffer) {
+
+    TransformedClass pre = Main.getInstance().getCurrentClass();
```

Comment:
can this be `getAndRemoveCurrentClass()`? it's nice to not leak the thread local beyond its scope

Reply:
1. Switched do JarOS everywhere.
2. I believe that we can remove after get so implemented as suggested.
3. The TL instance is virtually static (singleton). The original code implemented all methods (and fields ofc) static, but I hated that and for better testability moved to singleton (which isn't my fav design pattern either ;) ).